### PR TITLE
Add downloader support for the new Jenkins flow

### PR DESCRIPTION
**Pull request by kimpaller**
This implements/updates the ff:
1. [add] download_info_txt (usage: nebula --help dl.info-txt) Download info.txt from a custom url_template and show all or selected fields in csv or dict format
2. [modify] download_boot_files (usage: nebula --help dl.bootfiles) Downloader support for boot_partition folders -  "HDL_PRs, Linux_PRs, HDL_latest_commit, Linux_latest_commit

### CODE REVIEW

File: nebula/tasks.py [hunk 1]
```diff
@@ -512,11 +519,46 @@ def download_generate_bootbin_map_file(
     print("Mapping file saved to mapping.txt")
 
 
+@task(
+    help={
+        "url": "Artifactory url to path with info.txt",
+        "field": "Field to show. Will show all if None.",
```
Comment (tagoylo) [inline nebula/tasks.py:525]: What fields are available?

File: nebula/tasks.py [hunk 2]
```diff
@@ -512,11 +519,46 @@ def download_generate_bootbin_map_file(
...
+        print(",".join(to_show.values()))
+    else:
+        print(to_show)
+
 dl = Collection("dl")
 dl.add_task(download_sdcard, "sdcard")
 dl.add_task(download_boot_files, "bootfiles")
 dl.add_task(download_generate_matlab_bootbins, "matlab_bootbins")
 dl.add_task(download_generate_bootbin_map_file, "bootbin_map")
+dl.add_task(download_info_txt, "info_txt")
```
Comment (tagoylo) [inline nebula/tasks.py:561]: I think it would help to add test for this, perhaps under test_cli?

File: nebula/downloader.py [hunk 3]
```diff
@@ -377,8 +417,17 @@ def _get_artifactory_file(self, filename, folder, ip, branch, addl, url_template
             raise Exception(
                 "No server IP or domain name specified. Must be defined in yaml or provided as input"
             )
-        # get url template base
-        url = gen_url(ip, branch, folder, filename, addl, url_template)
+
+        new_flow = False
+        for pipeline in ["HDL_PRs","Linux_PRs", "HDL_latest_commit", "Linux_latest_commit"]:
```
Comment (tagoylo) [inline nebula/downloader.py:422]: Add test to downloader if new flow.

**Comment by tagoylo** [comment]
Fix lint

File: nebula/tasks.py (hunk 1 above)
Comment (kimpaller) [reply nebula/tasks.py:525]: built_projects, BRANCH,PR_ID,TIMESTAMP,DIRECTION,Triggered by, COMMIT SHA, COMMIT_DATE

File: nebula/tasks.py (hunk 2 above)
Comment (kimpaller) [reply nebula/tasks.py:565]: https://github.com/sdgtt/nebula/pull/112/commits/caa565edace45d5e16e2f7c0bb4d38a0bec85a3d

File: nebula/downloader.py (hunk 3 above)
Comment (kimpaller) [reply nebula/downloader.py]: https://github.com/sdgtt/nebula/pull/112/commits/caa565edace45d5e16e2f7c0bb4d38a0bec85a3d

**Comment by kimpaller** [comment]
> Fix lint

Lint fixed

**Review by tagoylo — APPROVED** [review]

**Pull request merged**